### PR TITLE
Move static database validation to only be executed on service start

### LINKED ISSUE #42845: Statically configured Oracle database breaks `tctl`
Expected behavior:

`tctl` works irrespective of configured databases

Current behavior:

With statically configured Oracle database in `teleport.yaml`:

```yaml
  - name: oracle
    protocol: oracle
    uri: localhost:2484
    static_labels:
      env: dev
```

`tctl` will complain about lack of enterprise license and refuse to start

```
tctl --debug edit user/tener
2024-06-12T17:05:50+02:00 DEBU             Debug logging has been enabled. common/tctl.go:287
2024-06-12T17:05:50+02:00 DEBU             reading config file: /etc/teleport.yaml config/configuration.go:397
ERROR: oracle database protocol is only available with an enterprise license
```

Bug details:
- Teleport version: v16 alpha
- Recreation steps: add static database to local Teleport configuration, issue `tctl` command which reads the local configuration
- Debug logs: see above.

### PR DESCRIPTION
**Pull request by gabrielcorado**
Closes #42845.

Given that the `ApplyConfig` initializes all the configuration and sets default values, it also invokes the `services.ValidateDatabase` present at the `servicecfg.Database.CheckAndSetDefaults`, performing a full database validation every time the configuration was loaded. We're removing this call. The static database validation is now done directly on the DB service initialization.

In addition to this change, this PR simplifies the `servicecfg.Database.CheckAndSetDefaults` implementation, removing the duplicated validation (AD) and tests.

changelog: Fix cases where `tctl` commands could not be executed due to static database configuration errors.

### CODE REVIEW

**Review by Tener — APPROVED** [review]
Some unit tests are failing, but looks good otherwise.

File: api/types/database.go [hunk 1]
```diff
@@ -922,6 +922,11 @@ func (d *DatabaseV3) CheckAndSetDefaults() error {
 		}
 	}
 
+	const defaultKRB5FilePath = "/etc/krb5.conf"
+	if d.Spec.AD.Domain != "" && d.Spec.AD.Krb5File == "" {
```
Comment (greedy52) [inline api/types/database.go:926]: Could you add a comment on why checking `d.Spec.AD.Domain !=""` here? (or move the old comment ?)

File: lib/service/servicecfg/database.go [hunk 2]
```diff
@@ -100,53 +93,22 @@ type OracleOptions struct {
...
-
 	// If AWS account ID is missing, but assume role ARN is given,
 	// try to parse the role arn and set the account id to match.
 	if d.AWS.AccountID == "" && d.AWS.AssumeRoleARN != "" {
 		parsed, err := awsutils.ParseRoleARN(d.AWS.AssumeRoleARN)
-		if err != nil {
-			return trace.BadParameter("database %q invalid AWS assume_role_arn: %v",
-				d.Name, err)
+		if err == nil {
+			d.AWS.AccountID = parsed.AccountID
```
Comment (greedy52) [inline lib/service/servicecfg/database.go:107]: Do you mind leaving a TODO to move this to API? We can either implement our own parser or use aws-sdk-go-v2

**Review by greedy52 — APPROVED** [review]

File: lib/config/configuration_test.go [hunk 3]
```diff
@@ -2847,12 +2801,8 @@ func TestDatabaseCLIFlags(t *testing.T) {
 				Name:     "sqlserver",
 				Protocol: defaults.ProtocolSQLServer,
 				URI:      "sqlserver.example.com:1433",
-				TLS: servicecfg.DatabaseTLS{
-					Mode: servicecfg.VerifyFull,
-				},
 				AD: servicecfg.DatabaseAD{
 					KeytabFile: "/etc/keytab",
-					Krb5File:   defaults.Krb5FilePath,
```
[the remark this answers is not in the document]
Comment (gabrielcorado) [reply lib/config/configuration_test.go:2855]: It is used as the default value of the `--ad-krb5-file` flag (`teleport db start`)

**Review by r0mant — APPROVED** [review]

**Comment by zmb3** [comment]
@gabrielcorado we want the backport label for v16 here, right?

**Pull request merged**